### ISSUE TITLE
Nil for 1 (hex) digit colors.

### ISSUE THREAD

**Issue by dominikoledzki** [thii/SwiftHEXColors#32]
For this call:
`UIColor(hex: color)`
Initializer returns nil for all values between 0x000000 and 0x00000f because of formatting string:
var hexString = String(format: "%2X", hex) (SwiftHEXColors.swift line 111) which results in
" 0" for 0x000000 etc.

First solution:
Change 
`var hexString = String(format: "%2X", hex)`
to
`var hexString = String(format: "%02X", hex)`

Second (better) solution:
You don't need to format Int to String, it's a waste of CPU time. The easiest method I can think of to convert Int to RGB color is:

```
convenience init(hex: Int) {
        let b = hex % 256
        let g = (hex >> 8) % 256
        let r = (hex >> 16) % 256

        let s = CGFloat(256)

        self.init(red: CGFloat(r)/s, green: CGFloat(g)/s, blue: CGFloat(b)/s, alpha: 1.0)
    }
```

**Comment by thii** [comment]
I'd vote for the second solution. Could you make a PR for that?

**Comment by viktorasl** [comment]
Hi @dominikoledzki . Thanks for reported issue, it was fixed with #33.
